google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java

```diff
@@ -185,22 +192,92 @@ public static TransportChannelProvider defaultTransportChannelProvider() {
         .setJwtEnabledScopes(JWT_ENABLED_SCOPES);
   }
 
-  /** Returns the object with the settings used for calls to ReadRows. */
+  /**
+   * Returns the object with the settings used for calls to ReadRows.
+   *
+   * <p>This is idempotent and streaming operation.
+   *
+   * <p>The idle timeout is configured via {@code ServerStreamingCallSettings#getIdleTimeout}. This
+   * timer will terminate any stream that has not has seen any demand or activity in the configured
+   * interval. To turn off idle checks, set the idleTimeout to {@link Duration#ZERO}.
+   *
+   * <p>Retries will be attempted if RPC failed with {@link Code#DEADLINE_EXCEEDED} and {@link
+   * Code#UNAVAILABLE} failure code.
+   *
+   * <p>RetryDelay starts with 100ms or 0.1 seconds, which multiplies with 1.3 times on every
+   * reattempt till a max retry timeout value of 60 seconds.
+   *
+   * <p>RPC timeout is 20 seconds.
+   *
+   * <p>Total timeout for this operation is 60 mins or 1 hour.
+   *
+   * @see RetrySettings for more explanation.
+   */
   public ServerStreamingCallSettings<Query, Row> readRowsSettings() {
     return readRowsSettings;
   }
 
-  /** Returns the object with the settings used for calls to SampleRowKeys. */
+  /**
+   * Returns the object with the settings used for calls to SampleRowKeys.
+   *
+   * <p>This is idempotent and non-streaming operation.
+   *
+   * <p>Retries will be attempted if RPC failed with {@link Code#DEADLINE_EXCEEDED} and {@link
+   * Code#UNAVAILABLE} failure code.
+   *
+   * <p>RetryDelay starts with 100ms or 0.1 seconds, which multiplies with 1.3 times on every
+   * reattempt till a max retry timeout value of 60 seconds.
+   *
+   * <p>RPC timeout is 20 seconds.
+   *
+   * <p>Total timeout for this operation is 600 seconds or 10 mins.
```

Comment:
Similar to the streaming example I think you should combine the timeouts:

The default timeout for  each attempt is 20 seconds and the timeout for the entire operation across all of the attempts is 10 mins.